lightning/src/ln/channelmanager.rs

```diff
@@ -3132,6 +3145,7 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 					msg: update
 				});
 			}
+			self.pending_events.lock().unwrap().push(events::Event::ChannelClosed { channel_id: msg.channel_id,  err: ClosureDescriptor::CooperativeClosure });
```

Comment:
Also I think we *should* split this into "CounterpartyInitiated" cooperative close, and "LocallyInitiated" cooperative close.

Reply:
Well I think we don't track initiator in `internal_closing_signed`, we could by adding yet-another-boolean in `Channel` but that's gross IMHO. Another solutions is actually putting the closure generation in `shutdown` but that's bouncing in a different direction that what your above comment is suggesting ?

Reply:
Leaving a TODO, I think we can improve on this feature once we have a bit of customer feedbacks and if that distinction is really important.